### DISCUSSION TITLE
Drive_Base with gyro on max speed

### DISCUSSION THREAD

**Discussion by Kermit647** [pybricks/support#1078] [General]
Hi,
i am using the current beta with gyro on Lego Spike.

When using DriveBase.drive with high speed and curve  (like drive(999,15), the steering will not work anymore. It will drive some curve, but not correct and will not react to turning the brick (gyro).
If I drive slower and leave some headroom so the duty cycle does not get close to 100% it will work. 
Unfortunatly the maximum safe speed will also go down with the battery voltage.

**Comment by laurensvalk** [comment]
Thanks for trying the new features! Can you give a full example script that exhibits your problem? 

We can only go as fast as the batteries allow, though.

Or are you suggesting to automatically slow down the speed in curves only but not in straights? That could theoretically work.

**Comment by Kermit647** [comment]
Steps to reproduce the problem :
Use a two-wheel robot and put it on some block so the wheels can move freely.
Now (with gyro enabled) use 
drivebase.drive(200,0).
The wheels will spin steadily. If you now turn the robot, you can hear the motors change speed, trying to compensate the spin.

Now use 
drivebase.drive(9999,15)
The wheels will start turning again. But if you now turn the robot, nothing happens, they will just go on with the same speed.

**Reply by laurensvalk** [reply]
Right, when attempting to drive at 10 m/s (which the robot can't) it doesn't have anything left to compensate for steering.

**Reply by laurensvalk** [reply]
But perhaps the question is, should it prioritize the turn rate over the drive speed?
